src/main/java/org/minions/devfund/noemiguzman/battleship/Ship.java

```diff
@@ -0,0 +1,217 @@
+        }
+    }
+
+    /**
+     * help to ask to place ship at.
+     * @param column numeric
+     * @param row numeric
+     * @param lenghtRow maxrow
+     * @param lengthColumn maxcolumn
+     * @param ocean object
+     * @return true
+     */
+    private boolean helpToPlaceShipAt(int column, int row, int lenghtRow, int lengthColumn, final Ocean ocean) {
+
+        for (int i = row - 1; i <= row + lenghtRow; i++) {
+            for (int j = column - 1; j < column + lengthColumn; j++) {
+                try {
+
+                    if (!ocean.getShipArray()[i][j].getShipType().equals("empty")) {
+                        return false;
+                    }
+                } catch (Exception e) {
+                    continue;
+                }
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Puts the ship in the ocean.
+     * This involves giving values to the bowRow, bowColumn, and horizontal instance variables in the ship.
+     * @param row number
+     * @param column number
+     * @param horizontal true
+     * @param ocean object
+     */
+    public void placeShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
+        bowRow = row;
+        bowColumn = column;
+        this.horizontal = horizontal;
+
+        /*
+         * Set references to the ship in the ships array in the Ocean object
+         */
+        if (horizontal) {
+            for (int j = column; j < column + getLength(); j++) {
+                ocean.getShipArray()[row][j] = this;
+            }
+        } else {
+            for (int i = row; i < row + getLength(); i++) {
+                ocean.getShipArray()[i][column] = this;
+            }
+        }
+    }
+
+
+    /**
+     * Return true if the part of the ship was hit, false otherwise.
+     * @param row number
+     * @param column number
+     * @return true
+     */
+    public boolean wasShootAt(int row, int column) {
+        if (horizontal) {
+            return hit[column - this.bowColumn];
+        } else {
+            return hit[row - this.bowRow];
+        }
+    }
+
+    /**
+     * If a part of the ship occupies the given row and column, and the ship hasnot been sunk,
+     * mark that part of the ship as hit (in the hit array, 0 indicates the bow) and return true,
+     * otherwise return false.
+     * @param row number
+     * @param column number
+     * @return true
+     */
+    public boolean shootAt(int row, int column) {
+        if (!isSunk()) {
+            if (horizontal) {
+                if (row == this.bowRow && column < this.bowColumn + length) {
+                    hit[column - this.bowColumn] = true;
+                    return true;
+                }
+            } else {
+                if (column == this.bowColumn && row < this.bowRow + length) {
+                    hit[row - this.bowRow] = true;
+                    return true;
+                }
+            }
+        }
+        return false;
```

Comment:
This may be refactored like this.
```java
    public boolean shootAt(int row, int column) {
        if (isSunk()) {
            return false;
        }
        final boolean isHittable = horizontal ? row == this.bowRow && column < this.bowColumn + length
                : column == this.bowColumn && row < this.bowRow + length;
        if (isHittable) {
            final int hitPosition = horizontal ? column - this.bowColumn : row - this.bowRow;
            hit[hitPosition] = true;
            return true;
        }
        return false;
    }
```